components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1073,6 +1082,40 @@ private OAuthResponse handleSuccessAuthorization(OAuthMessage oAuthMessage, OIDC
         return oauthResponse;
     }
 
+    private Optional<TokenBinder> getTokenBinder(String clientId) throws OAuthSystemException {
+
+        OAuthAppDO oAuthAppDO;
+        try {
+            oAuthAppDO = OAuth2Util.getAppInformationByClientId(clientId);
+        } catch (IdentityOAuth2Exception | InvalidOAuthClientException e) {
+            throw new OAuthSystemException("Failed to retrieve OAuth application with client id: " + clientId, e);
+        }
+
+        if (oAuthAppDO == null || StringUtils.isBlank(oAuthAppDO.getTokenBindingType())) {
+            return Optional.empty();
+        }
+
+        OAuth2Service oAuth2Service = getOAuth2OSGiService();
```

Comment:
you can use EndpointUtil.getOAuth2Service()

Reply:
fixed